### ISSUE TITLE
[Bug]dev install with npm 7.6 (idk the exact ver that breaks it)

### ISSUE THREAD

**Issue by Robonau** [manga-download/hakuneko#3184]
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Describe the bug**
`npm install` fails when using npm 7.6 or higher

**To Reproduce**
1. clone HN
2. `npm install` with 7.6 
3. it fails 

**Expected behavior**
works

**Screenshots**
see EvEr4EnGeL's installation on discord

**Context (please complete the following information):**
 - OS: [e.g. Windows10]
 - HakuNeko Version [e.g. 6.17]

**Additional context**
at least warn future devs by saying something in the [dev wiki](https://github.com/manga-download/hakuneko/wiki/Developer-Manual#3-install-dependencies)

ive had this issue come up on another persons install too (on non debian linux)

**Comment by 27136773** [comment]
I'm not sure if this is something that can be solved in HakuNeko project...
Could not be reproduced on macOS

**Comment by Sheepux** [comment]
Quoting myself from the discord:

> Anyway, it's not a node 15/npm 7.6 issue, it's something else
> what i mean is that it's not about the binary themselves, it could be about something they now configure with the installer

The issue is on a "fresh install" (meaning the user never installed node/npm).
I'm using the latest (updated from a previous version) and i don't have the issue.

Issue was related to node-gyp being present and failing.

```
gyp ERR! find VS looking for Visual Studio 2015
npm ERR! gyp ERR! find VS - not found
npm ERR! gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
```

The installer has a checkbox to install the "tools for native modules" (not checked  by default).
I think that i did check it in the past and that's probably the dependencies required.

**Comment by Robonau** [comment]
> Quoting myself from the discord:
> 
> > Anyway, it's not a node 15/npm 7.6 issue, it's something else
> > what i mean is that it's not about the binary themselves, it could be about something they now configure with the installer
> 
> The issue is on a "fresh install" (meaning the user never installed node/npm).
> I'm using the latest (updated from a previous version) and i don't have the issue.

problem here is that i 
uninstalled npm (command line)
installed newest version 7.6 (command line)
it failed to npm install in HN

if you are on npm 7.6 and try to npm install in HN it will fail

even if this cant be fixed warning ppl about this behaviour in the dev wiki is a good idea

as it is right now the instructions are incorrect if you are on 7.6
(we couldn't get EvEr4EnGeL's to npm install until he downgraded his npm version)
(can be reproduced on both pc and linux)

**Comment by Sheepux** [comment]
Let me rephrase that differently:
- node/npm now installs gyp by itself and i didn't get any issues with uninstalling/reinstalling node/npm (i'm currently on 15/7.6 all the time)
- you should probably check the additional checkbox for the "extra native tools" because i prorbably did check it in the past. Unless someone tries it we don't know if that's the thing to "warn".

**Comment by Robonau** [comment]
> Let me rephrase that differently:
> 
> * node/npm now installs gyp by itself and i didn't get any issues with uninstalling/reinstalling node/npm (i'm currently on 15/7.6 all the time)
> * you should probably check the additional checkbox for the "extra native tools" because i prorbably did check it in the past. Unless someone tries it we don't know if that's the thing to "warn".

if u install through command line there is no checkbox
all linux users will install via command line

you also upgraded to 7.6, you didn't install 7.6 
the issue is specific to first time installs
(hence me having to uninstall to reproduce the issue)

**Comment by Sheepux** [comment]
..  i'm dropping any debate on that one. Have fun with the rest.

**Comment by Robonau** [comment]
> .. i'm dropping any debate on that one. Have fun with the rest.
sheep realising he is wrong, so he throws the whole conversation in the gutter

**Comment by Sheepux** [comment]
What's the point in throwing in your personal flaming ? I'm not realising i'm wrong as i'm quite sure of the issue as stated before.
There is a dependency required that is done on the addiotional tool. I did test it by uninstalling and reinstalling and it was working, so i know what the issue is. All we have to find is the HOW (checkbox or additional commands).
If you still want to be a debating pain, feel free to do as ever.

**Comment by Robonau** [comment]
if that IS the issue then mention that in the dev wiki then
im currently re installing node again(checking the box that installs chocolatey, the only checkbox in the installer(this is not something ive ever done before)), to see if it is the issue

**Comment by 27136773** [comment]
If possible i would like to locate the dependency module that wants to be compiled on windows instead of providing ready to use binaries. Maybe we can get rid of the module...

I hope it is not electron itself ^^"

**Comment by Robonau** [comment]
the issue always seemed to be gyp related 
but i dont think we don't get rid of gyp

**Comment by Robonau** [comment]
ok so, it is todo with the install chocolatey+other stuff checkbox

edit:
nvm my npm version didn't change (idk still)

**Comment by Robonau** [comment]
- chocolatey-dotnetfx.extension v1.0.1
 - kb3033929 v1.0.5
 - visualstudio2019buildtools v16.9.1.0
 - python3 v3.9.2
 - chocolatey-windowsupdate.extension v1.0.4
 - vcredist140 v14.28.29910
 - kb2999226 v1.0.20181019
 - visualstudio-installer v2.0.1
 - kb2919355 v1.0.20160915
 - kb2919442 v1.0.20160915
 - chocolatey-visualstudio.extension v1.9.0
 - vcredist2015 v14.0.24215.20170201
 - dotnetfx v4.8.0.20190930
 - visualstudio2019-workload-vctools v1.0.1
 - kb3035131 v1.0.3
 - python v3.9.2
 
 should be all that is different from the checkbox
 if sheep is right it is one of these causing the issue

**Comment by 27136773** [comment]
The package that caused the trouble is [discord-rpc](https://www.npmjs.com/package/discord-rpc), or to be more specific one of its optional dependencies [register-scheme](https://www.npmjs.com/package/register-scheme).
The package has 2 Stars on GitHub and hasn't been updated in 3 years :(

**Simplified steps to reproduce:**
1. Install NodeJS with NPM > 7.x
2. Create a new directrory and change into it
3. Run `npm install register-scheme` or `npm install discord-rpc`

So for now if discord-rpc support should be kept, python and visual studio tools are required on the developer system to build this stupid package

**Comment by Robonau** [comment]
so tell new devs that python and visual studio tools need to be installed?

**Comment by 27136773** [comment]
I have to go back to work now and would appreciate if someone could verify that the discord status is still working with the PR
